OneSignalSDK/onesignal/src/main/java/com/onesignal/PushRegistratorFCM.java

```diff
@@ -92,9 +97,28 @@ private void initFirebaseApp(String senderId) {
       FirebaseOptions firebaseOptions =
          new FirebaseOptions.Builder()
             .setGcmSenderId(senderId)
-            .setApplicationId("OMIT_ID")
-            .setApiKey("OMIT_KEY")
+            .setApplicationId(getAppId())
+            .setApiKey(getApiKey())
+            .setProjectId(getProjectId())
             .build();
       firebaseApp = FirebaseApp.initializeApp(OneSignal.appContext, firebaseOptions, FCM_APP_NAME);
    }
+
+   private static @NonNull String getAppId() {
+      if (OneSignal.remoteParams.fcmParams.appId != null)
+         return OneSignal.remoteParams.fcmParams.appId;
+      return FCM_DEFAULT_APP_ID;
+   }
+
+   private static @NonNull String getApiKey() {
+      if (OneSignal.remoteParams.fcmParams.apiKey != null)
+         return OneSignal.remoteParams.fcmParams.apiKey;
+      return FCM_DEFAULT_API_KEY;
+   }
+
+   private static @NonNull String getProjectId() {
+      if (OneSignal.remoteParams.fcmParams.projectId != null)
+         return OneSignal.remoteParams.fcmParams.projectId;
+      return FCM_DEFAULT_PROJECT_ID;
+   }
```

Comment:
Please note that these values have to align, i.e. belong together:
The API key has to authenticate the project, identified by the Project ID.
The Application ID has to identify an application that belongs to the same project.

If the remote parameters do not match like this, requests will fail in the background,.